docs/userguide/storagedriver/device-mapper-driver.md

```diff
@@ -0,0 +1,306 @@
+
+To write 56KB of new data to a container:
+
+1. An application makes a request to write 56KB of new data to the container.
+
+2. The allocate-on-demand operation allocates a single new 64KB block to the containers snapshot.
+
+    If the write operation is larger than 64KB, multiple new blocks are allocated to the container snapshot.
+
+3. The data is written to the newly allocated block.
+
+#### Overwriting existing data
+
+To modify existing data for the first time:
+
+1. An application makes a request to modify some data in the container.
+
+2. A copy-on-write operation locates the blocks that need updating.
+
+3. The operation allocates new blocks to the container snapshot and copies the data into those blocks.
+
+4. The modified data is written into the newly allocated blocks.
+
+The application in the container is unaware of any of these
+allocate-on-demand and copy-on-write operations. However, they may add latency
+to the application's read and write operations.
+
+## Configuring Docker with Device Mapper
+
+The `devicemapper` is the default Docker storage driver on some Linux
+distributions. This includes RHEL and most of its forks. Currently, the following distributions support the driver:
+
+* RHEL/CentOS/Fedora
+* Ubuntu 12.04          
+* Ubuntu 14.04          
+* Debian  
+
+Docker hosts running the `devicemapper` storage driver default to a
+configuration mode known as `loop-lvm`. This mode uses sparse files to build
+the thin pool used by image and container snapshots. The mode is designed to work out-of-the-box
+with no additional configuration. However, production deployments should not run
+under `loop-lvm` mode.
+
+You can detect the mode by viewing the `docker info` command:
+
+    $ sudo docker info
+    Containers: 0
+    Images: 0
+    Storage Driver: devicemapper
+     Pool Name: docker-202:2-25220302-pool
+     Pool Blocksize: 65.54 kB
+     Backing Filesystem: xfs
+     ...
+     Data loop file: /var/lib/docker/devicemapper/devicemapper/data
+     Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
+     Library Version: 1.02.93-RHEL7 (2015-01-28)
+     ...
+
+The output above shows a Docker host running with the `devicemapper` storage driver operating in `loop-lvm` mode. This is indicated by the fact that the `Data loop file` and a `Metadata loop file` are on files under `/var/lib/docker/devicemapper/devicemapper`. These are loopback mounted sparse files.
+
+### Configure direct-lvm mode for production
+
+The preferred configuration for production deployments is `direct lvm`. This
+mode uses block devices to create the thin pool. The following procedure shows
+you how to configure a Docker host to use the `devicemapper` storage driver in a
+`direct-lvm` configuration.
+
+> **Caution:** If you have already run the Docker daemon on your Docker host and have images you want to keep, `push` them Docker Hub or your private Docker Trusted Registry perform attempting this procedure.
```

Comment:
*to* Docker Hub

perform attempting this procedure. -> before attempting this procedure?